subsys/bluetooth/services/fast_pair/fmdn/state.c

```diff
@@ -1295,14 +1298,12 @@ int fp_fmdn_state_init(void)
 		return err;
 	}
 
-	/* Create advertising set if beacon is provisioned. */
-	if (is_provisioned) {
-		if (!fmdn_adv_set) {
-			err = fmdn_adv_set_setup();
-			if (err) {
-				LOG_ERR("FMDN State: fmdn_adv_set_setup failed: %d", err);
-				return err;
-			}
+	/* Reserve the FMDN advertising set only in the Fast Pair enabled state. */
```

Comment:
I also wonder if we still need to handle case without `fmdn_adv_set` in `bt_fast_pair_fmdn_adv_param_set`. Now we can return error on `fast_pair_enable` in similar way as we do it for TX power (that could improve consistency of error handling and slightly simplify implementation).

Reply:
> I also wonder if we still need to handle case without `fmdn_adv_set` in `bt_fast_pair_fmdn_adv_param_set`. Now we can return error on `fast_pair_enable` in similar way as we do it for TX power (that could improve consistency of error handling and slightly simplify implementation).

It limits the API functionality

Reply:
> `bt_fast_pair_fmdn_tx_power_verify` no longer needs to create and delete set (it should reuse the set that was allocated on init I think).

It is called in the `state_init` function to validate the TX power configuration. This function is well-contained and it does need to reuse the `fmdn_adv_set`.

Reply:
> Just noticed that there is no need to check if we are provisioned in this function (we assign `is_provisioned = fp_fmdn_state_is_provisioned();`, but then do nothing with `is_provisioned` in scope of this function since we introduced `fmdn_post_init_work_handle`)

well, that's weird that I don't get any build warning regarding unused variables. I will fix this part :)

Reply:
> > I also wonder if we still need to handle case without `fmdn_adv_set` in `bt_fast_pair_fmdn_adv_param_set`. Now we can return error on `fast_pair_enable` in similar way as we do it for TX power (that could improve consistency of error handling and slightly simplify implementation).
> 
> It limits the API functionality

Ok so we can stick to early validation of the advertising parameters.